xcube/api/gen/gen.py

```diff
@@ -254,10 +256,36 @@ def step8(dataset):
         if append_mode and os.path.exists(output_path):
             # noinspection PyShadowingNames
             def step9(dataset):
-                output_writer.append(dataset, output_path, **output_writer_params)
+                global _APPEND_DS_TO_DC
+                if output_path.endswith('.nc'):
+                    output_writer.append(dataset, output_path, **output_writer_params)
+                elif output_path.endswith('.zarr'):
+                    _APPEND_DS_TO_DC = check_append_or_insert(time_range, output_path)
+                    if _APPEND_DS_TO_DC is True:
+                        output_writer.append(dataset, output_path, **output_writer_params)
+                    elif _APPEND_DS_TO_DC is False:
+                        tempdir_for_merging = os.path.join(os.path.dirname(__file__), '..', '..', '..',
```

Comment:
Oh nooo, this is really bad!

Use OS-specific temporary location, use https://docs.python.org/3.7/library/tempfile.html